### ISSUE TITLE
The meaning of description in modification

### ISSUE THREAD

**Issue by eshmoylova** [modelica/ModelicaSpecification#3397]
The syntax of modification allows to specify a description-string on a modification of an element:
```
element-modification :
   name [ modification ] description-string
```
How is that description string supposed to be interpreted? In a redeclaration, it is clear that it becomes the description-string of the element. But what about the case without `redeclare`? For example:
```
model M1
  parameter Integer n "description 1";
...
end M1;
model M2
  extends M1(n(min = 1) = 1 "description 2");
end M2;
```
Is it expected that `M2.n` will have "description 2" or "description 1"?

**Comment by HansOlsson** [comment]
I don't see that it is specified, and Dymola uses "description 1".
TL;DR: I would propose that we standardize that and non-normatively add that it is rarely used.

To me there are two possible uses of that description:

- Better description of the _parameter_. Logically it makes sense, but I have some difficulty finding good use cases for this.
- A description of the _modifier_; e.g., `extends M1(n(min = 1) = 4 "Computed based on measurements in June");` If we further modify `n` this new description should then be gone.

Note that the choices-annotation uses this exact syntax - _for descriptions of the modifiers_. The specification has an example:
```
type KindOfController = Integer(min = 1, max = 3)
  annotation(choices(
              choice = 1 "P",
              choice = 2 "PI",
              choice = 3 "PID"));
```
Having that replace the description of the parameter would be confusing for me, and having two different description seems like a bit of over-design (especially as we would need to figure out how to set both initially).

**Comment by henrikt-ma** [comment]
In Wolfram System Modeler, `b.x` gets the description `This is B` in this case:
```
model M
  model A
    parameter Real p = 0 "This is A";
  end A;
  model B
    extends A(p "This is B");
  end B;
  B b;
end M;
```

**Comment by HansOlsson** [comment]
Adding to the point about choices.

The specification https://specification.modelica.org/master/inheritance-modification-and-redeclaration.html#annotation-choices-for-suggested-redeclarations-and-modifications actually contains `A a(x = 3 "PID");` where the description text should is linked to the parameter _value_ and not to the parameter itself.

**Comment by henrikt-ma** [comment]
The thing with `choices` is that it is an annotation; the information that the annotation needs to convey needs to be arranged in the form of _class-modification_ in the grammar; it doesn't make the annotation an actual class modification.

**Comment by HansOlsson** [comment]
> The thing with `choices` is that it is an annotation; the information that the annotation needs to convey needs to be arranged in the form of _class-modification_ in the grammar; it doesn't make the annotation an actual class modification.

Assume we want description in modifiers to override descriptions (seems like there's agreement).
The question can then be formulated as follows: what modifier should we generate from a choice-annotation with description?
The cases above indicate that no description is likely the best choice, since the descriptions in the choice-annotations describe the choices - not a change of description of the component.

**Comment by henrikt-ma** [comment]
@HansOlsson, your own reaction in the poll isn't one of the four alternatives.

**Comment by HansOlsson** [comment]
> @HansOlsson, your own reaction in the poll isn't one of the four alternatives.
> 
> Edit: I suppose the poll should say _hooray_ instead of _pizza slice_? (I plan to vote for option 2 once it's clear that _hooray_ corresponds to this alternative.)

Yes. That was the intent; they should have clearer symbols.

**Comment by eshmoylova** [comment]
Adding _eyes_ because I feel like the offered options do not consider whether or not the modification is used on an instance declaration or in an extends clause. The example I had was specifically about the extends clauses. It makes sense that the description on an instance does not override the description on a variable or a parameter whether it is specified directly or comes from a choice annotation. But it would make sense to me that in case on an extends clause (whether or not there is a redeclare present) that the description is actually a description of the variable. In this case, the descriptions in choices also fall into the categories of whether it is a modification for an instance or an extends clause (typically a redeclaration).

**Comment by HansOlsson** [comment]
Phone meeting: accepted variant 2.
